src/Libraries/DynamoUnits/Utilities.cs

```diff
@@ -35,8 +67,95 @@ public static double ParseExpression(string expression)
             return ForgeUnitsEngine.parseUnitless(expression);
         }
 
+        /// <summary>
+        /// Returns a formatted unit value string using unitType and unitSymbol type ids.
+        /// Sample: '12.345 m3' is a value of '12.345', of unit 'meter', of symbol 'meter cubed (m3)', of number format 'decimal', of precision '3'.
+        /// </summary>
+        /// <param name="numValue"></param>
+        /// <param name="unitTypeId"></param>
+        /// <param name="unitSymbolId"></param>
+        /// <param name="precision"></param>
+        /// <param name="decimalFormat"></param>
+        /// <returns></returns>
+        [IsVisibleInDynamoLibrary(false)]
+        public static string ReturnFormattedString(double numValue, string unitTypeId, string unitSymbolId, int precision, string numberFormat)
```

Comment:
Yeah it was. But we need to to be a string because of the following. Basically we need to build a function call and for that we need the input to be either string, int, double, bool...:
`public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
        {
            if (!InPorts[0].IsConnected)
            {
                return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), AstFactory.BuildNullNode()) };
            }


            var functionNode = AstFactory.BuildFunctionCall(new Func<double, string, string, int, string, string>(DynamoUnits.Utilities.ReturnFormattedString), 
                new List<AssociativeNode> { inputAstNodes[0], AstFactory.BuildStringNode(SelectedUnit.TypeId),
                AstFactory.BuildStringNode(SelectedSymbol.TypeId), AstFactory.BuildIntNode(SelectedPrecision), AstFactory.BuildStringNode(SelectedFormat.ToString())});
            return new[] { AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), functionNode) };
        }`

Reply:
ah, i see